fuel-core/src/tx_pool.rs

```diff
@@ -37,6 +37,8 @@ pub enum Error {
     Database(Box<dyn StdError>),
     #[error("unexpected block execution error {0:?}")]
     Execution(crate::executor::Error),
+    #[error("Tx is invalid, insertion failed {0:?}")]
+    String(String),
```

Comment:
This doesn't preserve the stack trace or other parts of the error context, however we could easily fix that with this:

```suggestion
    Other(#[from] anyhow::Result),
```